### PR TITLE
DCPR Requests to Metadata records after CSI approval

### PR DESCRIPTION
**Pull request by Samweli**
Fixes https://github.com/kartoza/ckanext-dalrrd-emc-dcpr/issues/218

Adds logic to handle mapping of DCPR requests to metadata records after an approval from CSI representative

Screenshot of a metadata record resulting from a DCPR request.
image

Updated look of the metadata records made from DCPR requests.
image

### CODE REVIEW

**Comment by gubuntu** [comment]
Thanks @Samweli. I see checks are failing but besides that, can you confirm this occurs upon CSI approval OR rejection. In either case, the metadata record must be published. 

Also please confirm that there is a field showing whether it was approved or rejected.

**Comment by Samweli** [comment]
@gubuntu it currently working only CSI approval. I'm working on including the rejection action and the field value for showing the original action on metadata record item.

File: ckanext/dalrrd_emc_dcpr/helpers.py [hunk 1]
```diff
@@ -342,3 +342,19 @@ def get_dcpr_requests_approved_by_nsif(request_origin):
         "dcpr_request_list_awaiting_csi_moderation"
     )({"request_origin": request_origin})
     return dcpr_requests_approved_by_nsif
+
+
+def is_dcpr_request(package):
+
+    for extra in package.get("extras") or []:
+        if extra.get("key") == "origin" and extra.get("value") == "DCPR":
```
Comment (Mohab25) [inline ckanext/dalrrd_emc_dcpr/helpers.py:350]: will this drop a keyError if extras is not in the package ?

Comment (Samweli) [reply ckanext/dalrrd_emc_dcpr/helpers.py:350]: @Mohab25 the `get` returns None if the key doesn't exist. It doesn't throw any error.

File: ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py [hunk 2]
```diff
@@ -271,11 +271,56 @@ def dcpr_request_csi_moderate(
...
+            ]:
+                create_package_from_dcpr_request(
+                    context, request_obj, moderation_action
+                )
     else:
         raise toolkit.ObjectNotFound
     return result
 
 
+def create_package_from_dcpr_request(
```
[the remark this answers is not in the document]
Comment (Samweli) [reply ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py:287]: @Mohab25 yes you are right, I was thinking about this see https://github.com/kartoza/ckanext-dalrrd-emc-dcpr/blob/ec0551da6a0d74cf3a123cd4f7d353164b149117/ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py#L295, will make the change.

[the remark this answers is not in the document]
Comment (Samweli) [reply ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py:288]: yes @Mohab25, you are right. I was actually thinking of moving the values to constants file before making the PR see . thanks. will make the change.

**Pull request merged**